### PR TITLE
[WFCORE-5106] Add configuration option in Elytron subsystem to change file encoding in FileAuditEndpoint.

### PR DESCRIPTION
**Pull request by ivassile**
https://issues.redhat.com/browse/WFCORE-5106
Note: This change depends on Elytron issue ELY-1211. Merge after PR https://github.com/wildfly-security/wildfly-elytron/pull/1900

https://issues.redhat.com/browse/EAP7-1557
Analysis document: https://github.com/wildfly/wildfly-proposals/pull/335

### CODE REVIEW

**Comment by ivassile** [comment]
Closed PR: https://github.com/wildfly/wildfly-core/pull/4319

File: elytron/src/main/java/org/wildfly/extension/elytron/AuditResourceDefinitions.java [hunk 1]
```diff
@@ -106,6 +107,12 @@ class AuditResourceDefinitions {
             .setRestartAllServices()
             .build();
 
+    static final SimpleAttributeDefinition ENCODING = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.ENCODING, ModelType.STRING, true)
+            .setAllowExpression(true)
```
Comment (OndrejKotek) [inline elytron/src/main/java/org/wildfly/extension/elytron/AuditResourceDefinitions.java:111]: As the new attribute allow expressions it should be tested with an expression set.

**Comment by Skyllarr** [comment]
@ivassile  Here's the PR for subsystem bump to version 18:

https://github.com/wildfly/wildfly-core/pull/5463

Please update this PR so it's based off of that one. Thanks!

**Comment by fjuma** [comment]
@ivassile This looks good, we just need this to be applied on top of the subsystem bump PR that Diana mentioned. Thanks!

**Review by fjuma — CHANGES_REQUESTED** [review]
@ivassile We should also add a test case to the audit logging tests (please check if the existing audit logging tests are in Core or WildFly and add an appropriate test there too). Thanks!

**Comment by ivassile** [comment]
@fjuma Updated the PR to be based on subsystem version 18. Created https://issues.redhat.com/browse/WFLY-18048 with a test.

**Review by Skyllarr — APPROVED** [review]

File: elytron/src/main/java/org/wildfly/extension/elytron/AuditResourceDefinitions.java [hunk 2]
```diff
@@ -106,6 +107,12 @@ class AuditResourceDefinitions {
             .setRestartAllServices()
             .build();
 
+    static final SimpleAttributeDefinition ENCODING = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.ENCODING, ModelType.STRING, true)
```
Comment (OndrejKotek) [inline elytron/src/main/java/org/wildfly/extension/elytron/AuditResourceDefinitions.java:110]: The default value should be set or documented in the descriptions of related attributes.

Comment (ivassile) [reply elytron/src/main/java/org/wildfly/extension/elytron/AuditResourceDefinitions.java:110]: Added description that the default is UTF_8.

**Comment by yersan** [comment]
@ivassile Elytron has been merged but I cannot kick off this one again, could you rebase and push it up again?

**Comment by yersan** [comment]
This is all progressing well, git hub should refresh later status of all the jobs

https://ci.wildfly.org/buildConfiguration/WFCorePRAggregator/372248?buildTab=dependencies&mode=timeline&type=snapshot

**Comment by yersan** [comment]
@ivassile notice this is going to conflict with #5463 since the commit hashes of similar changes are different, to avoid it, you could rebase on top of it.

**Comment by darranl** [comment]
@ivassile this contains the wrong commits for the Elytron subsystem bump still, it needs to contain the exact same versions as in Farah's PR

**Review by fjuma — APPROVED** [review]
Thanks @ivassile!

Thanks for adding a test as well! Instead of the new PR, I think it would be better to just include that commit in the [existing PR](https://github.com/wildfly/wildfly/pull/13543) with the WildFly docs (you can still reference WFLY-18048 there) so everything for this feature for WildFly is in one place.

**Comment by fjuma** [comment]
@yersan @darranl Just FYI, the commit SHAs are correct now. Thanks @ivassile!

**Comment by ivassile** [comment]
@fjuma Included the test commit in PR https://github.com/wildfly/wildfly/pull/13543

**Comment by darranl** [comment]
Please don't make any further changes to this PR - I am going to combine it with a couple of other PRs so we can get them merged together.

Thank you for this implementation.

**Pull request merged**